Source/Scene/Model/GeoJsonLoader.js

```diff
@@ -495,12 +541,154 @@ function parse(geoJson, frameState) {
 
   const primitive = new ModelComponents.Primitive();
   primitive.attributes = attributes;
-  primitive.indices = indices;
   primitive.featureIds = featureIds;
-  primitive.primitiveType = PrimitiveType.LINES;
+  primitive.primitiveType = PrimitiveType.POINTS;
   primitive.material = material;
 
-  const primitives = [primitive];
+  return primitive;
+}
+
+function parse(geoJson, frameState) {
+  const result = new ParseResult();
+
+  // Parse the GeoJSON
+  const parseFunction = geoJsonObjectTypes[geoJson.type];
+  if (defined(parseFunction)) {
+    parseFunction(geoJson, result);
+  }
+
+  const features = result.features;
+  const featureCount = features.length;
+
+  if (featureCount === 0) {
+    throw new RuntimeError("GeoJSON must have at least one feature");
+  }
+
+  // Allocate space for property values
+  const properties = {};
+  for (let i = 0; i < featureCount; i++) {
+    const feature = features[i];
+    const featureProperties = defaultValue(
+      feature.properties,
+      defaultValue.EMPTY_OBJECT
+    );
+    for (const propertyId in featureProperties) {
+      if (featureProperties.hasOwnProperty(propertyId)) {
+        if (!defined(properties[propertyId])) {
+          properties[propertyId] = new Array(featureCount);
+        }
+      }
+    }
+  }
+
+  // Fill in the property values. Default to empty string for undefined values.
+  for (let i = 0; i < featureCount; i++) {
+    const feature = features[i];
+    for (const propertyId in properties) {
+      if (properties.hasOwnProperty(propertyId)) {
+        const value = defaultValue(feature.properties[propertyId], "");
+        properties[propertyId][i] = value;
+      }
+    }
+  }
```

Comment:
Is there a reason this is done as two passes? it looks like they could be combined.

Reply:
It doesn't know the full list of properties until it has looped over every feature. Once it knows that then it can start assigning values in the second loop.